nmigen/vendor/xilinx_s7.py

```diff
@@ -0,0 +1,254 @@
+from abc import abstractproperty
+
+from ..hdl.ast import *
+from ..hdl.dsl import *
+from ..hdl.ir import *
+from ..build import *
+
+
+__all__ = ["XilinxS7Platform"]
+
+
+class XilinxS7Platform(TemplatedPlatform):
```

Comment:
Nit: can we call this `Xilinx7SeriesPlatform`, since that's what used in all official docs? The file would be renamed as well.